### PR TITLE
[FEATURE] Use SuiteFactory API in `tests/actions`

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by cdkini — APPROVED** [review]

File: tests/actions/test_validation_operators.py [hunk 1]
```diff
@@ -105,16 +106,16 @@ def warning_failure_validation_operator_data_context(
     data_context.update_expectation_suite(expectation_suite=warning_expectations)
 
     failure_expectations.expectation_suite_name = "f2.failure"
-    data_context.add_expectation_suite(expectation_suite=failure_expectations)
+    data_context.suites.add(ExpectationSuite(name=failure_expectations))
```
Comment (tyler-hoffman) [inline tests/actions/test_validation_operators.py]: Looks like this and the next 3 are passing an ExpectationSuite as the name, unless I'm missing something.

File: tests/actions/test_validation_operators_in_data_context.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def validation_operators_data_context(
     warning_expectations = df.get_expectation_suite(discard_failed_expectations=False)
 
     failure_expectations.expectation_suite_name = "f1.failure"
-    data_context.add_expectation_suite(expectation_suite=failure_expectations)
+    data_context.suites.add(ExpectationSuite(name=failure_expectations))
```
Comment (tyler-hoffman) [inline tests/actions/test_validation_operators_in_data_context.py]: Same as above for this and the next

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]
Looks to me like we're passing an instance of a suite as a name in 6 places.

File: tests/actions/test_validation_operators.py (hunk 1 above)
Comment (joshua-stauffer) [reply tests/actions/test_validation_operators.py]: i think this is actually correct, the signature is `SuiteFactory.add(suite: ExpectationSuite)` ([source](https://github.com/great-expectations/great_expectations/blob/4ce453be08c43c36b8b7c6a77fe1eaada8821ee1/great_expectations/core/suite_factory.py#L24))

Comment (tyler-hoffman) [reply tests/actions/test_validation_operators.py]: I think the error is the argument to ExpectationSuite, not the factory's add. `name=failure_expectations`

Comment (joshua-stauffer) [reply tests/actions/test_validation_operators.py]: ah, thanks, you're right!

**Comment by joshua-stauffer** [comment]
this PR requires functionality in https://github.com/great-expectations/great_expectations/pull/9364

**Review by tyler-hoffman — APPROVED** [review]

**Pull request merged**